repository.go

```diff
@@ -255,6 +255,20 @@ func (v *Repository) SetHeadDetached(id *Oid, sig *Signature, msg string) error
 	return nil
 }
 
+func (v *Repository) IsHeadDetached() (bool, error) {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_repository_head_detached(v.ptr)
+	switch ret {
+	case 1:
+		return true, nil
+	case 0:
+		return false, nil
+	}
+	return false, MakeGitError(ret)
```

Comment:
Our style is to check for `ret < 0` and then do an early return with the error, we can then simply perform a `ret != 0` check to convert the value into a boolean, which match the semantics of C, rather than just 1 or 0.